plot/CanonicalUtterance.java

```diff
@@ -0,0 +1,104 @@
+import java.util.*;
+
+public class CanonicalUtterance
+{
+	String path;
+	String value;
+	public CanonicalUtterance() { }
+	//path should be seperated by period, e.g. encoding.y.timeUnit
+	public CanonicalUtterance(String path, String value) {
+		this.path = path;
+		this.value = value;
+	}
+
+
+	public String get_simple_canonical_utterance()
+	{     
+		String first_path_key = "";  
+		if(path.split("\\.").length > 0)  {
+		    first_path_key = path.split("\\.")[0];
+         }	
+		if(first_path_key.equals("encoding") || first_path_key.equals("layer") || first_path_key.equals("config") || first_path_key.equals("spec"))
+		{   
+			int startPath = first_path_key.length() + 1;
+			path = path.substring(startPath);
+		}
+		String u = "Set "+path+" to "+value+".";
+		return u;
+
+	}
+
+
+	public String get_canonical_utterance()
+	{
+		String u = "";
+		if(path.equals("encoding.x.scale.nice"))
```

Comment:
Better to define a data structure that contains these and then use a for loop over it...when possible, decouple data from control.  For example: String[][] items = { Item("encoding.x.scale.nice", "Set scale"), ... }